z3c/dependencychecker/modules.py

```diff
@@ -0,0 +1,398 @@
+# -*- coding: utf-8 -*-
+from xml.etree import ElementTree
+from z3c.dependencychecker.dotted_name import DottedName
+import ast
+import fnmatch
+import os
+import re
+
+
+TEST_REGEX = r"""
+{0}    # Needs to start with the path separator
+\w*    # There could be some characters (or none)
+test   # has to contain test on it
+s?     # ensure plurals are handled as well
+(      # Needs to end with either a path or a file extension:
+{0}    # path separator
+|      # OR
+\w*    # some possible characters
+\.     # a dot
+\w+    # extension
+)
+""".format(os.sep)
+
+TEST_IN_PATH_RE = re.compile(
+    TEST_REGEX,
+    re.VERBOSE,
+)
+
+
+class BaseModule(object):
+
+    def __init__(self, package_path, full_path):
+        self.path = full_path
+        self._relative_path = self._get_relative_path(package_path, full_path)
+        self.testing = self._is_test_module()
+
+    @staticmethod
+    def _get_relative_path(package_path, full_path):
+        return full_path[len(package_path):]
+
+    def _is_test_module(self):
+        return bool(re.search(TEST_IN_PATH_RE, self._relative_path))
+
+    @classmethod
+    def search_files(cls, top_dir):
+        raise NotImplementedError
+
+    def scan(self):
+        raise NotImplementedError
+
+
+class PythonModule(BaseModule):
+
+    @classmethod
+    def search_files(cls, top_dir):
```

Comment:
`def instantiate_from_package_dir()` ?

Reply:
See above my large answer, although still open to change them if you still think the name change is needed.

Reply:
"search" doesn't sound so very active to me. I'd like something with "create" or "instantiate", that makes it clearer that this is the main object-instantiating method.

What about `create_from_files()`? Then `scan()` can stay. You create objects from files and then ask them to scan themselves. Sounds clear enough.